### PR TITLE
SEARCH-342 Fix includeAllFields handling

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/IResearch/Search.cpp [hunk 1]
```diff
@@ -139,23 +139,41 @@ std::shared_ptr<Index> getIndex(LogicalCollection const& collection,
...
-auto Find(SearchMeta::Map const& map, V&& v) noexcept {
-  return map.find(std::forward<V>(v));
+auto Find(SearchMeta::Map const& map, std::string_view key) noexcept {
+  return map.find(key);
 }
 
-template<typename V>
-auto Find(auto const& vector, V const& v) noexcept {
+auto Find(auto const& vector, std::string_view key) noexcept {
   auto last = end(vector);
```
Comment (gnusi) [inline arangod/IResearch/Search.cpp:147]: `std::end`

File: arangod/IResearch/Search.cpp [hunk 2]
```diff
@@ -139,23 +139,41 @@ std::shared_ptr<Index> getIndex(LogicalCollection const& collection,
...
   auto it = std::lower_bound(
-      begin(vector), last, v,
+      begin(vector), last, key,
```
Comment (gnusi) [inline arangod/IResearch/Search.cpp:149]: `std::begin`

File: arangod/IResearch/Search.cpp [hunk 3]
```diff
@@ -260,6 +264,55 @@ void add(SearchMeta& search, IResearchInvertedIndexMeta const& index) {
 
 }  // namespace
 
+struct MetaFst : VectorFst {};
+
+std::shared_ptr<SearchMeta> SearchMeta::make() {
+  return std::make_shared<SearchMeta>();
```
Comment (gnusi) [inline arangod/IResearch/Search.cpp:270]: NIT
really need this? :-)

**Review by gnusi — CHANGES_REQUESTED** [review]

File: arangod/IResearch/Search.h [hunk 4]
```diff
@@ -41,6 +43,8 @@ namespace arangodb::iresearch {
 class SearchFactory;
 class IResearchInvertedIndex;
 
+struct MetaFst;
+
 struct SearchMeta final {
```
Comment (gnusi) [inline arangod/IResearch/Search.h:48]: `class`?

File: arangod/IResearch/Search.h [hunk 5]
```diff
@@ -50,6 +54,18 @@ struct SearchMeta final {
   };
   using Map = std::map<std::string, Field, std::less<>>;
   Map fieldToAnalyzer;
+
+  [[nodiscard]] static std::shared_ptr<SearchMeta> make();
+
+  void createFst();
```
Comment (gnusi) [inline arangod/IResearch/Search.h:60]: do we need this as `public` method?

File: arangod/IResearch/Search.cpp [hunk 6]
```diff
@@ -645,9 +698,13 @@ Result Search::updateProperties(CollectionNameResolver& resolver,
                 }
                 return error;
               });
-  if (r.ok()) {
-    _meta = std::move(searchMeta);
+  if (!r.ok()) {
+    return r;
   }
+  if (ServerState::instance()->isSingleServer()) {
```
Comment (Dronplane) [inline arangod/IResearch/Search.cpp:704]: Don't we need the same on the DBServer?

File: arangod/IResearch/Search.cpp (hunk 3 above)
Comment (MBkkt) [reply arangod/IResearch/Search.cpp:270]: Yes, incomplete type

File: arangod/IResearch/Search.h (hunk 5 above)
Comment (MBkkt) [reply arangod/IResearch/Search.h:60]: Yes

File: arangod/IResearch/Search.cpp (hunk 6 above)
Comment (MBkkt) [reply arangod/IResearch/Search.cpp:704]: We do, check viewnode

**Review by Dronplane — APPROVED** [review]
LGTM

**Review by gnusi — APPROVED** [review]

**Pull request merged**